src/plugins/integrations/PostgreIntegration.ts

```diff
@@ -41,66 +43,69 @@ class PostgreIntegration implements Integration {
           });
-        }
 
-        const pgQuery = query.apply(this, arguments);
+          if (params) {
+            span.addTags({
+              [SpanTags.SPAN_TYPE]: SpanTypes.RDB,
+              [DBTags.DB_INSTANCE]: params.database,
+              [DBTags.DB_USER]: params.user,
+              [DBTags.DB_HOST]: params.host,
+              [DBTags.DB_PORT]: params.port,
+              [DBTags.DB_TYPE]: DBTypes.PG,
+              [SpanTags.TOPOLOGY_VERTEX]: true,
+              [SpanTags.TRIGGER_DOMAIN_NAME]: LAMBDA_APPLICATION_DOMAIN_NAME,
+              [SpanTags.TRIGGER_CLASS_NAME]: LAMBDA_APPLICATION_CLASS_NAME,
+              [SpanTags.TRIGGER_OPERATION_NAMES]: [tracer.functionName],
+            });
+          }
 
-        let statement = pgQuery.text;
-        statement = Utils.replaceArgs(statement, pgQuery.values);
+          const pgQuery = query.apply(this, arguments);
 
-        if (statement) {
-          const statementType = statement.split(' ')[0].toUpperCase();
-          span.addTags({
-            [DBTags.DB_STATEMENT_TYPE]: statementType,
-            [DBTags.DB_STATEMENT]: statement,
-            [SpanTags.OPERATION_TYPE]: SQLQueryOperationTypes[statementType] ? SQLQueryOperationTypes[statementType] : 'READ',
-          });
-        }
+          let statement = pgQuery.text;
+          statement = Utils.replaceArgs(statement, pgQuery.values);
 
-        const originalCallback = pgQuery.callback;
-
-        pgQuery.callback = (err: any, res: any) => {
-          if (err) {
-            const parseError = Utils.parseError(err);
-            span.setTag('error', true);
-            span.setTag('error.kind', parseError.errorType);
-            span.setTag('error.message', parseError.errorMessage);
+          if (statement) {
+            const statementType = statement.split(' ')[0].toUpperCase();
+            span.addTags({
+              [DBTags.DB_STATEMENT_TYPE]: statementType,
+              [DBTags.DB_STATEMENT]: statement,
+              [SpanTags.OPERATION_TYPE]: SQLQueryOperationTypes[statementType] ? SQLQueryOperationTypes[statementType] : 'READ',
+            });
           }
 
-          span.close();
+          const originalCallback = pgQuery.callback;
 
-          if (originalCallback) {
-            originalCallback(err, res);
-          }
-        };
+          pgQuery.callback = (err: any, res: any) => {
+            if (err) {
+              const parseError = Utils.parseError(err);
+              span.setTag('error', true);
+              span.setTag('error.kind', parseError.errorType);
+              span.setTag('error.message', parseError.errorMessage);
+            }
 
-        return pgQuery;
+            span.close();
 
+            if (originalCallback) {
+              originalCallback(err, res);
+            }
+          };
+
+          return pgQuery;
+        } catch (error) {
+          ThundraLogger.getInstance().error(error);
+          query.apply(this, arguments);
```

Comment:
Can we be sure that in case of error `query.apply` is not called twice

Reply:
In case of error in the wrapped function error callback is called. I verified, it is not called twice.